src/features/filter/filterSlice.js

```diff
@@ -0,0 +1,47 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const INITIAL_STATE = {
+  filteredTagsInHome: [],
+  filteredTagsInProfile: [],
+};
+
+const filterSlice = createSlice({
+  name: "filter",
+  initialState: INITIAL_STATE,
+  reducers: {
+    tagSelectedInHome: (state, action) => {
+      if (state.filteredTagsInHome.indexOf(action.payload) === -1) {
```

Comment:
aren't we already checking this while dispatching?

Reply:
Yes @ray-wtag bhai. So, we don't need to check here again.I updated this.